spec/javascripts/govuk_publishing_components/analytics-ga4/pii-remover.spec.js

```diff
@@ -320,6 +320,110 @@ describe('GOVUK.analyticsGa4.PIIRemover', function () {
     })
   })
 
+  function generatePhoneNumber (prefix, amount, suffix = '') {
+    var phoneNumber = prefix
+    for (var i = 0; i < amount; i++) {
+      phoneNumber += Math.floor(Math.random() * 10)
+    }
+    return phoneNumber + suffix
+  }
+
+  describe('phone numbers', function () {
+    it('are stripped if they are valid UK mobile numbers', function () {
+      // Test 50 random phone numbers so we aren't storing any numbers in code.
+      for (let i = 0; i < 50; i++) {
+        const phoneNumber = generatePhoneNumber('07', 9)
+        const redactedNumber = pii.stripPIIWithOverride(phoneNumber)
+        expect(redactedNumber).toEqual('[phone number]')
+      }
+    })
+
+    it('are stripped if they are valid international UK mobile numbers', function () {
+      // Test 50 random phone numbers so we aren't storing any numbers in code.
+      for (let i = 0; i < 50; i++) {
+        const phoneNumber = generatePhoneNumber('+447', 9)
+        const redactedNumber = pii.stripPIIWithOverride(phoneNumber)
+        expect(redactedNumber).toEqual('[phone number]')
+      }
+    })
+
+    it('are stripped if they are valid UK landline numbers', function () {
+      // Test 50 random phone numbers so we aren't storing any numbers in code.
+      for (let i = 0; i < 50; i++) {
+        // A landline can start with 01, 02, 04, or 06, so we generate all 4 examples here.
+        const numbers = [
+          generatePhoneNumber('01', 9),
+          generatePhoneNumber('02', 9),
+          generatePhoneNumber('04', 9),
+          generatePhoneNumber('06', 9),
+          generatePhoneNumber('01', 8),
+          generatePhoneNumber('02', 8),
+          generatePhoneNumber('04', 8),
+          generatePhoneNumber('06', 8)]
+        for (const number of numbers) {
+          const redactedNumber = pii.stripPIIWithOverride(number)
+          expect(redactedNumber).toEqual('[phone number]')
+        }
+      }
+    })
+
+    it('are stripped if they are valid international UK landline numbers', function () {
+      // Test 50 random phone numbers so we aren't storing any numbers in code.
+      for (let i = 0; i < 50; i++) {
+        // A landline can start with 01, 02, 04, or 06, so we generate all 4 examples here.
+        const numbers = [generatePhoneNumber('+441', 9), generatePhoneNumber('+442', 9), generatePhoneNumber('+444', 9), generatePhoneNumber('+446', 9)]
+        for (const number of numbers) {
+          const redactedNumber = pii.stripPIIWithOverride(number)
+          expect(redactedNumber).toEqual('[phone number]')
+        }
+      }
+    })
+
+    it('are not stripped if they are invalid UK numbers', function () {
+      const numbers = [
+        generatePhoneNumber('07', 7),
```

Comment:
Might be worth documenting the details of these numbers like you have with the other tests. Presumably any number starting with `07` followed by 7 digits is invalid?

Side thought - is it possible the rules for phone numbers could change in future?